### ISSUE TITLE
Subpanels not working in version 6.1.0

### ISSUE THREAD

**Issue by hybridvision** [FrDH/jQuery.mmenu#751]
Hi Fred,

First of all, thanks for a very useful plugin :)

I've discovered a bug that seems to have appeared in recent versions and is present in the latest 6.1.0 release: subpanels no longer work properly.

A live example is better than words: https://codepen.io/anon/pen/owBvgX

As you'll see in the CodePen, there are some odd display issues with the subpanel and when clicking on a link to open it, it appears briefly and then everything disappears.

If I go back to using version 5.7.8, it works as expected with no other changes to code.

I'm on OS X 10.10.5 and have seen the same bug in Chrome, Firefox and Safari.

Thanks for your time,

Stephen

**Comment by FrDH** [comment]
Hi,

There seems to be an issue with panels in the new approach, thanks for bringing it to my attention.
I'll see if this can be fixed shortly.

For now, you could try to move the second panel out of the first, and place it at the same level. The classname "Panel" should be sufficient. Have a look at this demo, it basically does the same as your codepen:
http://mmenu.frebsite.nl/demo/index.html?demo=app-settings

Hope this helps.

**Comment by hybridvision** [comment]
Thanks for the fast reply and the workaround! I'll try it first thing on Monday when I get to the office :)

Have a good weekend!

**Comment by hybridvision** [comment]
Hello again,

I tried your workaround and unfortunately it still exhibits the same buggy behaviour.

At first, I moved the sub-panel onto the same level as the first panel (they were sibling divs) but when I opened the menu, I only saw the contents of the sub-panel, which was the last div inside the mmenu container.

I found that if I wrap both the main panel and the sub-panel in another div, then it renders both panels but the bug is the same as before. I prepared an updated example here: https://codepen.io/anon/pen/WOpRyB

Maybe I misunderstood your workaround so would you mind having a look at the new example?

I also studied the link you provided and it looks like the sub-panels are all nested within their parent panels, rather than being called separately like in my version.

For now, I'm back to using version 5.7.8 but I'd like to use some of the newer options from 6.x if possible.

Thanks for your time on this!

**Comment by FrDH** [comment]
You should not wrap both the main and the sub panel in another DIV, the first solution should have worked. I'm not sure why it didn't. I'll see if I can get the codePen to work.

**Comment by hybridvision** [comment]
I thought it was odd to have to wrap both the main and sub panel in another div but if I don't, this is what I get: https://codepen.io/anon/pen/qjrevJ - only the sub panel shows when I open the menu...

**Comment by FrDH** [comment]
I think I made it work: 
https://codepen.io/anon/pen/JJNPGx

What I changed:
- removede the #wrapper DIV
- used the navbars add-on (the default navbar back button and title only work with standard LI > UL structure)
- added HTML for "navbar information"

If you don't want this approach (with a fixed navbar and extra HTML), you could also use the standard LI > UL structure, nut this would mean all anchors to a subpanel need to be in a listview and will be styled as such:
https://codepen.io/anon/pen/mwmbmV

**Comment by hybridvision** [comment]
Thanks very much for your help - it's much clearer now and I've been rethinking how I will approach this.

I'm working to integrate MMenu into a complex existing application where the menu structure is generated via JS and needs to be flexible based on configuration settings in the app. Once it is confirmed that it can work with our app, we'll be purchasing a commercial license :)

As part of trying to get a better understanding of what was happening, I tested it further and realised that starting from my last example (https://codepen.io/anon/pen/qjrevJ), the only change I had to make was the remove the `class="Panel"` from the sub panel. Then it works as expected. It seems that there's a definite bug there...

I should also add that my original example code came from the docs: http://mmenu.frebsite.nl/tutorials/advanced/the-menu.html - this sub panel code doesn't work anymore unless the sub panel is not nested in the parent and the `Panel` class is removed. I guess this is also a bug but if it's not, then the docs need to be updated to reflect this.

Lastly, another documentation comment: it was only by chance that I discovered I could add the class `mm-nopanel` to stop some child content being turned into a menu (some of my sub menu content contains ULs). It would be great if that was added to the docs as I'm sure I'm not the only one who was caught by it.

I know it's difficult maintaining a project like this and keeping all the docs up to date so this is by no means a criticism. Thanks again for all your work!

**Comment by FrDH** [comment]
> the only change I had to make was the remove the class="Panel" from the sub panel. Then it works as expected. It seems that there's a definite bug there...
The part where panels are created (including the navbars and listviews) was completely rewritten in 6.1. All seemed to work but I guess I didn't test this scenario. The issue actually is pretty obvious: the plugin first refactors all classnames "Panel" to "mm-panel" and then test for the classname "mm-panel" so it won't create a panel twice... This should be the other way around.

Thanks for noting the docs need to be updated, I'll do that shortly.

In general: if something isn't in the documentation, you probably shouldn't use it as its usage might change without notification or fallback. That being said, I think "mm-nopanel" should have been in the docs somewhere :)

**Comment by hybridvision** [comment]
I appreciate your explanation of what caused the bug - interesting to know :)

You make a very good point about not using things that aren't documented but I figured this was pretty safe since it was something you suggested here: https://github.com/FrDH/jQuery.mmenu/issues/499#issuecomment-170649868 - maybe this could be the basis for the documentation update.

Another one I came across that might be worth documenting is the `mm-noanimation` class for disabling the animation on a panel.

Cheers!
